src/Features/CSharp/Portable/InitializeParameter/CSharpAddParameterCheckCodeRefactoringProvider.cs

```diff
@@ -92,5 +93,22 @@ protected override StatementSyntax CreateParameterCheckIfStatement(DocumentOptio
                 statement: ifTrueStatement,
                 @else: null);
         }
+
+        protected override Document? TryAddNullCheckToParameterDeclaration(Document document, ParameterSyntax parameterSyntax)
+        {
+            var tree = parameterSyntax.SyntaxTree;
+            var options = (CSharpParseOptions)tree.Options;
+            if (options.LanguageVersion < LanguageVersionExtensions.CSharpNext)
+            {
+                return null;
+            }
+
+            // We expect the syntax tree to already be in memory since we already have a node from the tree
+            var syntaxRoot = tree.GetRoot();
```

Comment:
so, in general, we would far prefer the CT just be passed along and used here.  that way we have no dependencies/assumptions we have to fulfill.

Reply:
Fixed.